Puyolib/AssetManager.h

```diff
@@ -7,29 +7,25 @@
 #include "DebugLog.h"
 #include "Frontend.h"
 #include "GameSettings.h"
+#include <atomic>
 #include <set>
 #include <string>
 namespace ppvs {
-
+class AssetManagerPriv;
 class AssetManager {
 public:
 	AssetManager();
-	AssetManager(Frontend* fe, DebugLog* dbg);
+	AssetManager(DebugLog* dbg);
 	~AssetManager();
 
-	// Creates an unactivated clone of the current manager
-	AssetManager* clone();
+	// Creates a rendering agent for the Manager
+	AssetManagerPriv* generate_priv(Frontend* frontend, DebugLog* m_dbg);
```

Comment:
Naming scheme has been corrected. The primary reason that wrapper object exists is to prevent `ppvs::Game` objects from handling the global Asset Manager. I use this to prevent accidentally manipulating with it for all other instances / Client unexpectedly and without thread safety. It also allows us not to type the frontend as an argument to all asset load calls within a single `Game` object.